rust/src/push/mod.rs

```diff
@@ -447,13 +450,26 @@ impl FilteredPushRules {
             .iter()
             .filter(|rule| {
                 // Ignore disabled experimental push rules
+
+                if !self.msc1767_enabled && rule.rule_id.contains("org.matrix.msc1767") {
+                    return false;
+                }
+
                 if !self.msc3664_enabled
                     && rule.rule_id == "global/override/.im.nheko.msc3664.reply"
                 {
                     return false;
                 }
 
-                if !self.msc1767_enabled && rule.rule_id.contains("org.matrix.msc1767") {
+                if !self.msc3381_polls_enabled
+                    && (rule.rule_id == "global/override/.org.matrix.msc3930.rule.poll_response"
+                        || rule.rule_id
+                            == "global/underride/.org.matrix.msc3930.rule.poll_start_one_to_one"
+                        || rule.rule_id == "global/underride/.org.matrix.msc3930.rule.poll_start"
+                        || rule.rule_id
+                            == "global/underride/.org.matrix.msc3930.rule.poll_end_one_to_one"
+                        || rule.rule_id == "global/underride/.org.matrix.msc3930.rule.poll_end")
+                {
```

Comment:
It looks like for MSC1767 we simplify a bit:

```suggestion
                if !self.msc3381_polls_enabled && rule.rule_id.contains("org.matrix.msc3930") {
```

Reply:
Oh, fair enough!